taichi/transforms/type_check.cpp

```diff
@@ -355,6 +391,32 @@ class TypeCheck : public IRVisitor {
     if (!matching) {
       error();
     }
+    // Insert assertions if debug is on
+    if (stmt->op_type == BinaryOpType::pow) {
+      if (is_integral(stmt->rhs->ret_type) &&
+          is_integral(stmt->lhs->ret_type)) {
+        if (config_.debug) {
+          auto iter = stmt->parent->find(stmt);
+          auto compare_rhs = Stmt::make<ConstStmt>(TypedConstant(0));
+          auto compare = std::make_unique<BinaryOpStmt>(
+              BinaryOpType::cmp_ge, stmt->rhs, compare_rhs.get());
+          compare->ret_type = PrimitiveType::i32;
+          if (iter == stmt->parent->statements.begin() ||
```

Comment:
Could you explain a bit about this condition?

Reply:
We insert an assertion when:
1. The stmt is the first one in the block
2. The stmt before `pow` is not an assertion
3. The stmt before `pow` is not the assertion checking the exponent

Reply:
Why do we check these conditions?

Reply:
Type check will run several times, so if we don't check these, there will be duplicated assertions inserted.

Reply:
This method seems a bit hacky - is it better to put these runtime assertions into a separate pass? wdyt @jim19930609

Reply:
Yup, `irpass::check_out_of_bound` sounds like a better place for this assertion.

Reply:
Good idea. I noticed that this pass will potentially run more than once:
https://github.com/taichi-dev/taichi/blob/f1068ae3d6265067d4614387bbb232967e9e994b/taichi/transforms/check_out_of_bound.cpp#L112-L119

the workaround can be less hacky tho